csrc/trtllm_fmha_kernel_launcher.cu

```
@@ -25,317 +28,194 @@
+  DISPATCH_PYTORCH_DTYPE_TO_CTYPE(query.scalar_type(), DTypeQ, [&]() {
+    return DISPATCH_PYTORCH_DTYPE_TO_CTYPE(key_value_cache.scalar_type(), DTypeKV, [&]() {
+      return DISPATCH_PYTORCH_DTYPE_TO_CTYPE(out.scalar_type(), DTypeO, [&]() {
+        int num_qo_heads = query.size(1);
+        int sum_seq_q = query.size(0);
+        int head_dim_qk = query.size(2);
+        int head_dim_vo = out.size(-1);
+        int max_num_blocks_per_seq = block_tables.size(-1);
+        int num_pages = key_value_cache.size(0);
+        // NOTE(Zihao): key_value_cache is [num_pages, 2, num_kv_heads, page_size, head_dim]
+        int page_size = key_value_cache.size(-2);
+        int num_kv_heads = key_value_cache.size(-3);
+
+        int kv_stride_0 = key_value_cache.stride(-2);
+        int kv_stride_1 = key_value_cache.stride(-3);
+        int kv_stride_2 = key_value_cache.stride(-4);
+
+        auto device = query.device();
+        const auto stream = at::cuda::getCurrentCUDAStream(device.index());
+
+        trtllm_paged_attention_launcher<DTypeQ, DTypeKV, DTypeO, TllmPagedAttentionMode::Context>(
+            static_cast<DTypeO*>(out.data_ptr()), static_cast<DTypeQ*>(query.data_ptr()),
+            static_cast<DTypeKV*>(key_value_cache.data_ptr()),
+            static_cast<DTypeKV*>(key_value_cache.data_ptr() +
+                                  key_value_cache.stride(1) * key_value_cache.element_size()),
+            workspace_buffer.data_ptr(), static_cast<KVCachePageIndex*>(block_tables.data_ptr()),
+            static_cast<int*>(seq_lens.data_ptr()), batch_size, max_q_len, max_kv_len, num_pages,
+            num_qo_heads, num_kv_heads, head_dim_qk, head_dim_vo, page_size, kv_stride_0,
+            kv_stride_1, kv_stride_2, max_num_blocks_per_seq, bmm1_scale, bmm2_scale, window_left,
+            sum_seq_q, sm_count, stream,
+            cum_seq_lens_q.defined() ? cum_seq_lens_q.data_ptr<int>() : nullptr,
```

@PerkzZheng Are `cum_seq_lens_q` and `cum_seq_lens_kv` required for Context phase attention kernels?

Asking because if they are required, we should just throw an error if users do not provide these